webpack/redux/reducers/RedHatRepositories/enabled.js

```diff
@@ -58,39 +59,46 @@ const changeRepoState = (state, repoToChange, stateDiff) => (
 );
 
 export default (state = initialState, action) => {
-  if (action.type === DISABLE_REPOSITORY_REQUEST) {
-    return changeRepoState(state, action.repository, { loading: true });
-  } else if (action.type === DISABLE_REPOSITORY_FAILURE) {
-    return changeRepoState(state, action.payload.repository, { loading: false });
-  } else if (action.type === ENABLED_REPOSITORIES_REQUEST) {
-    if (!action.silent) {
-      return state.set(['loading'], true);
+  switch (action.type) {
+    case DISABLE_REPOSITORY_REQUEST:
+      return changeRepoState(state, action.repository, { loading: true });
+    case DISABLE_REPOSITORY_FAILURE:
+      return changeRepoState(state, action.payload.repository, { loading: false });
+    case ENABLED_REPOSITORIES_REQUEST:
+      if (!action.silent) {
+        return state.set(['loading'], true);
+      }
+      return state;
+    case ENABLED_REPOSITORIES_SUCCESS: {
+      const {
+        page,
+        per_page, // eslint-disable-line camelcase
+        subtotal,
+        results,
+      } = action.response;
+      return Immutable({
+        repositories: mapRepositories(results),
+        pagination: {
+          page: Number(page) || 1,
+          // server can return per_page: null when there's error in the search query,
+          // don't store it in such case
+          // eslint-disable-next-line camelcase
+          perPage: Number(per_page || state.pagination.perPage),
+        },
+        itemCount: Number(subtotal),
+        loading: false,
+        searchIsActive: !isEmpty(action.search),
+        search: action.search,
+      });
     }
-  } else if (action.type === ENABLED_REPOSITORIES_SUCCESS) {
-    const {
-      page, per_page, subtotal, results, // eslint-disable-line camelcase
-    } = action.response;
-
-    return Immutable({
-      repositories: mapRepositories(results),
-      pagination: {
-        page: Number(page) || 1,
-        // server can return per_page: null when there's error in the search query,
-        // don't store it in such case
-        // eslint-disable-next-line camelcase
-        perPage: Number(per_page || state.pagination.perPage),
-      },
-      itemCount: Number(subtotal),
-      loading: false,
-      searchIsActive: !isEmpty(action.search),
-      search: action.search,
-    });
-  } else if (action.type === ENABLED_REPOSITORIES_FAILURE) {
-    return Immutable({
-      repositories: [],
-      loading: false,
-      error: action.error,
-    });
+    case ENABLED_REPOSITORIES_FAILURE:
+      return Immutable({
+        repositories: [],
+        loading: false,
+        error: get(action, ['error']),
+        missingPermissions: parseMissingPermissions(get(action, ['result', 'response', 'data', 'error', 'details'])),
```

Comment:
I prefer readable code, how about something like:

``` js 
const errors = get(action, ['result', 'response', 'data', 'error', 'details']);
// some code

missingPermissions: parseMissingPermissions(errors);
```

I prefer that the reducer will get the filtered data already, so this logic IMHO should be in the action.